lib/interactive_viewer.ml

```diff
@@ -104,6 +104,64 @@ let change_summary z_patches : ui Lwd.t =
   in
   W.string ~attr:Notty.A.(fg lightcyan) operation_count
 
+
+(** Side by side diff view implementation **)
+
+type view_mode = SideBySide | Normal
+
+let view_mode = Lwd.var SideBySide
+
+let toggle_view_mode () =
+  match Lwd.peek view_mode with
+  | Normal -> Lwd.set view_mode SideBySide
+  | SideBySide -> Lwd.set view_mode Normal
+
+let rec split_and_align_hunk hunks mine_acc their_acc =
+  match hunks with
+  | [] -> (List.rev mine_acc, List.rev their_acc)
+  | `Common _ as common :: t ->
+      split_and_align_hunk t (common :: mine_acc) (common :: their_acc)
+  | `Mine s :: t ->
+      split_and_align_hunk t (`Mine s :: mine_acc) (`Common "" :: their_acc)
+  | `Their s :: t ->
+      split_and_align_hunk t (`Common "" :: mine_acc) (`Their s :: their_acc)
+
+let lines_to_ui_with_numbers lines attr_line_number attr_change =
```

Comment:
The line numbers complicate a bit the PR and are a great feature on their own. I'd like to make sure we have a solid implementation of the feature before we add line numbers.

What do you think of leaving line numbers out of this PR for now ? I've opened this issue: https://github.com/panglesd/diffcessible/issues/49

Reply:
Ok. Resolved.